coccinelle: 1.0.0-r23 -> 1.0.6

dtzWill opened the pull request:
Remove camlp4 dependency, documentation
says it is no longer needed.

Keeping pycaml however so it uses ours
instead of the bundled version.

Enable tests for sanity.

###### Motivation for this change


###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

dtzWill committed:
coccinelle: 1.0.0-r23 -> 1.0.6

* Remove camlp4 dependency, no longer needed
* Keep pycaml to use ours over bundled version
* Enable tests for sanity (broken on darwin)
* no longer need to force specific ocaml version
* don't create python wrappers, they don't seem to be needed
* remove deprecated configure flag.
* fix spgen by removing bytecode version.

vbgl commented:
Tests fail on darwin:

> Testing if spatch works with python scripts...
> Python error: dlopen(/nix/store/jjvnm28scpgrbm9pklmrk5373k9g2scf-python-2.7.13/lib/python2.7/lib-dynload/operator.so, 2): Symbol not found: __PyUnicodeUCS4_AsDefaultEncodedString
>  Referenced from: /nix/store/jjvnm28scpgrbm9pklmrk5373k9g2scf-python-2.7.13/lib/python2.7/lib-dynload/operator.so
>  Expected in: flat namespace
> in /nix/store/jjvnm28scpgrbm9pklmrk5373k9g2scf-python-2.7.13/lib/python2.7/lib-dynload/operator.so

Maybe it’s OK to just disable tests on darwin.

Also, restricting OCaml to version 4.01 seems no longer needed (you can drop the corresponding line in `all-packages.nix`).

dtzWill commented:
Thanks @vbgl I've updated with your suggestions.  I also dropped the generation of the python wrappers as AFAICT they aren't needed any longer? As mentioned in the commit the wrapper was added (with doubt shortly thereafter) in 2012 so it seems likely this is indeed no longer needed.  Even so, any thoughts on the matter are appreciated (I'm not sure how to test the python bits other than poking at 'pycocci') and please re-test on Darwin when you get a chance.

dtzWill commented:
Eep, looks like 'spgen' needs a patchelf or something--it refers to the /lib/ld-linux interpreter. 'spgen.opt' is fine though:

```console
$ file result/lib/coccinelle/spgen/spgen*
result/lib/coccinelle/spgen/spgen:     ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.24, BuildID[sha1]=a3bf1fc28c6611d0dce213eae47df8a02bfdf397, not stripped
result/lib/coccinelle/spgen/spgen.opt: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /nix/store/68sa3m89shpfaqq1b9xp5p1360vqhwx6-glibc-2.25/lib/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, not stripped
```

dtzWill commented:
(Done for now, rebased to current master)

FRidh commented on pkgs/development/tools/misc/coccinelle/default.nix:
```diff
@@ -1,36 +1,37 @@
-  # I don't have a non-NixOS system, so I cannot verify this, but shouldn't
-  # libpython know where to find its modules? (the path is for example in
-  # its Sys-module).
-  postInstall =
```
that's indeed silly. Python should always be able to find its own module...

dtzWill replied:
Great, thanks for the confirmation!

FRidh commented:
I have no comments on the contents of the PR except that the commits needs to be squashed.

dtzWill commented:
rebased + squashed.

grahamc commented:
Applied in 9614ebf137, thank you!
